### PR TITLE
meson: Minor improvements

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

File: src/boot/efi/meson.build [hunk 1]
```diff
@@ -349,8 +349,7 @@ foreach archspec : efi_archspecs
 endforeach
 
 foreach efi_elf_binary : efi_elf_binaries
-        # FIXME: Use build_tgt.name() with meson >= 0.54.0
-        name = fs.name(efi_elf_binary.full_path()).split('.')[0]
+        name = efi_elf_binary.name()
```
Comment (poettering) [inline src/boot/efi/meson.build:352]: hmm, the commit msg doesn't really match what this code does?

Comment (keszybz) [reply src/boot/efi/meson.build:352]: It does, no?

**Comment by keszybz** [comment]
#23391 does some of the same stuff, but it needs a rebase. We can merge this now.

File: src/boot/efi/meson.build (hunk 1 above)
Comment (medhefgo) [reply src/boot/efi/meson.build:352]: `typeof(efi_elf_binary) == build_tgt`

Comment (poettering) [reply src/boot/efi/meson.build:352]: ah

**Comment by poettering** [comment]
lgtm

**Review by bluca — APPROVED** [review]

**Pull request merged**